_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+
+Another possible solution is to bake in social consensus to the L1 contract, so that if a smart contract change is to be made, it first needs to be agreed by all the participants of the L2 network. However, this opens up to new problems regarding this type of governance, such as the handling of urgent situations, voter fatigue, and power centralization[^1]<sup>,</sup>[^2]. It would also be advisable to add delays, so if the users don’t like an upgrade, they have time to exit, and move their assets back to the L1. 
+
+
+## Why Proof of Stake (PoS) Might Not Work
+
+Currently, zk-rollups have a centralization in the block creator. To decentralize the block creator, a question is raised on who we decide as the leader / coordinator for a given block. 
+
+There are multiple answers to this question (some we will discuss further in this article): MEV Auction, Burn Auction, Proof of Efficiency and finally Proof of Stake. Most know Proof of Stake from their implementation in L1 blockchains such as Ethereum. However, they can also be used in zk-rollups for leader election, with some teams actively working on this: Polygon Zero, and Starkware. 
+
+The problem with integrating PoS into a zk rollup is that you are unable to fo a hard fork, which provided most of the security guarantees in L1 implementations. This makes it susceptible to DOS and Fake DOS attacks. 
+
+A DOS attack happens when a sufficiently large group of colluding validators with a lot of stake tries to leverage it by minting empty blocks or being extremely exclusive with the transactions they include. This basically allows them to have full control over the transactions in the rollup. They could:
+
+
+
+1. Charge high fees in order to include transactions in the block. 
+2. Prevent withdrawals. If they have a short position on the network token of the rollup, this could give them economic rewards. 
+3. Lock up supplies of certain tokens. This allows them to reap economic benefits from liquidity pools of said tokens. 
+
+In addition to these DOS attacks, there is also a Slashing attack to consider. This is possible because of the slashing condition that punishes stakers who do not create a block within a specified time X. If a block producer refuses to include transactions from other block producers, even with blocks containing their own transactions, in favor of uncling them the stakers will get slashed. If the block producer conducting this attack has a combined active stake with at least a ⅓ of active voting power, then this could be an effective, with-standing attack. Otherwise, it would be fairly rare for a small staker to get two blocks in a row, so it would not have much effect in the long-run. 
```

Comment:
I am not fully knowledgable of this form of attack. I will contact Daniel, and make the appropriate changes. Thank you for raising.

Reply:
Got the responses from Daniel.
1. This would be considered a griefing attack, in that there isn't an obvious benefit to the attacker, but the attacker could cause harm to their target. I.e. by censoring all blocks from a certain validator, the attacker can deny that validator their rewards, and potentially cause them to be slashed, if the protocol slashes validators for failing to produce blocks. 
2. An uncle block is a block that the following validators didn't build on top of, so it doesn't become part of the canonical chain; it just lives on an abandoned fork. In this scenario the "uncling" would be intentional/nefarious, but in other cases it can happen just because a block didn't propagate through the network quickly enough
3. Most PoS chains use BFT consensus to confirm each block,  so a block must be signed by 2/3 of voters (i.e. validators) in order to be confirmed. Thus, an attacker with 1/3 voting power could veto/censor any block they like. They could just ignore the block, and the network would eventually give up on trying to confirm it, resulting in a skipped/empty block.

Reply:
add to footnote?